riocli/config/config.py

```diff
@@ -81,6 +83,7 @@ def new_client(self, with_project: bool = True) -> Client:
 
         return Client(auth_token=token, project=project)
 
+    @lru_cache(maxsize=2)
```

Comment:
question: Why is the size set to 2?

Reply:
So `lru_cache` caches the output based on the arguments passed to the function. 

In this case, we have a boolean `with_project`, which means we can have a `True` or a `False` argument. The cache map will have two entries for these two possibilities. If we set it to `1`, then `lru_cache` will clear it whenever something with `with_project=False` is invoked in the same flow.

Reply:
Got it, can we add a comment to explain this as well.